src/containers/NonBitGoRecovery/Coin/Coin.tsx

```diff
@@ -73,26 +101,289 @@ export function Coin() {
         />
       );
     case 'eth':
-    case 'teth':
+    case 'gteth':
       return (
         <EthereumForm
+          onSubmit={async (values, { setSubmitting }) => {
+            setSubmitting(true);
+            try {
+              const bitGoEnvironment = coin === 'eth' ? 'prod' : 'test';
+              await window.commands.setBitGoEnvironment(
+                bitGoEnvironment,
+                values.apiKey
+              );
+              const chainData = await window.queries.getChain(coin);
+
+              const { gasLimit, maxFeePerGas, maxPriorityFeePerGas, ...rest } =
+                values;
+              if (
+                Number(gasLimit) <= 0 ||
+                Number(gasLimit) !== parseInt(gasLimit, 10)
```

Comment:
`Number(x)` and `parseInt(x, 10)` are equivalent so this condition will never pass